scriptlets/wait_for_snap_complete

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 Canonical Ltd.
+#
+# This script waits for all snap operations on a remote device to complete.
+# It uses SSH to query the snap changes on the device and checks for any ongoing operations.
+# The script will retry up to 40 times, with a 30-second sleep between each attempt.
+# This script is intended to be a drop-in replacement for inline convenience
+# functions previously defined in each of the Jenkins jobs.
+#
+# Dependencies:
+#     This script depends on SSH and assumes that SSH options and DEVICE_IP are set.
+#
+# Usage:
+#     Run directly:
+#     `wait_for_snap_complete <command> [command_args ...]`
+#     Or source it in another script and call
+#     `wait_for_snap_complete_function`
+
+wait_for_snap_complete_function() {
+    source "$(dirname "$BASH_SOURCE")/set-ssh-options"
+    set +x
+    loopcnt=0
+    until [ $loopcnt -gt 40 ]
+    do
+        loopcnt=$((loopcnt+1))
+        changes="$(ssh $SSH_OPTS $DEVICE_IP 'sudo snap changes')"
+        echo Checking for all snap operations to complete...
+        echo $changes
+        if [ -n "$changes" ] &&
+            [[ "$changes" != *"Doing"* ]] &&
+            [ "$(ssh $SSH_OPTS $DEVICE_IP echo true)" = "true" ]; then
+            done="true"
+            break
+        else
```

Comment:
minor nitpick on things like this, where you can just as well skip the else and just do
```
...
    fi
    sleep 30
done
```